lib/TutorialServo/TutorialServo.cpp

```diff
@@ -0,0 +1,34 @@
+
+#include "TutorialServo.h"
+
+TutorialServo::TutorialServo(PinName servoPin, float servoRangeInDegrees, float minPulsewidthInMs,
+                float maxPulsewidthInMs): m_servoPwmOut(servoPin),
+                              m_servoRangeInDegrees(servoRangeInDegrees), 
+                              m_minPulsewidthInMs(minPulsewidthInMs),
+                              m_maxPulsewidthInMs(maxPulsewidthInMs) {
+  // setting the member variables using initializer list
+  // otherwise nothing to do
+}
+
+TutorialServo::~TutorialServo() {
+  m_servoRangeInDegrees = 0;
+  m_minPulsewidthInMs   = 0;
+  m_maxPulsewidthInMs   = 0;
+  // felt i should include a destructor although seems trivial
+}
+
+void TutorialServo::setAngleRangeInDegrees(float degrees) {
+  m_servoRangeInDegrees = degrees;
+  // updating the member variable not sure if there is anything else to do here
+}
+
+void TutorialServo::setPulsewidthRangeInMs(float minPulsewidthMs, float maxPulsewidthMs) {
+  m_minPulsewidthInMs = minPulsewidthMs;
+  m_maxPulsewidthInMs = maxPulsewidthMs;
+  // updating the member variables (not sure if there is anything else to do here)
+}
+
+void TutorialServo::setPositionInDegrees(float degrees) {
```

Comment:
Here, you take an extra step. you have values in ms but need to pass in values in us so you multiply by 1000. See if you can avoid using us in your calculations at all.

Reply:
Also, the math doesnt really checkout. what if the range is 90?

Reply:
Changed it to ms, but not sure what you mean by the math is wrong. I added a check to see if degrees are greater than the servo range.

Reply:
i dont wanna make it too obvious but try plugging in values. Given these unrealistic but techincally valid parameters {minpw = 1, maxpw = 2, degrees = 90, range = 90} I expect to get an output of 1.5. But, with your code, I get 2.5 which is incorrect.

Reply:
@younesr1 For this the `period_ms(int ms)` function does not work because the parameter is interger type so you cannot specify values between milliseconds (ex. 1.5ms or 50% duty cycle).

`period_us(int us)` and `period(float seconds)` functions are ok

Reply:
good catch alex, i thought it'd take in a float. @Yehia-Fahmy  you can disregards what i had said about taking an extra step. what you had before was fine, my bad. However, my comment about the math not checking out still stands.

Reply:
i called alex to clear up my confusions about the pulsewidth function, let me know if this looks good